### PR TITLE
major speedup when drawing frame

### PR DESCRIPTION
**Pull request by neophob**
lunch break improvement: rendering of frame is about 5 times faster!

### CODE REVIEW

File: wasm/main.js [hunk 1]
```diff
@@ -7,21 +7,18 @@
...
-        lastPixel = pixel;
-        ctx.fillStyle = '#' + pixel.toString(16).padStart(6, '0') + 'FF';
-      }
-      const x = offset % 320;
-      const y = parseInt(offset / 320, 10);
-      ctx.fillRect(x, y, 1, 1)
-      offset++;
+    const pixel = tempImage.data;
+    let ofs = 0;
+    buffer.forEach((p) => {
```
Comment (pfiaux) [inline wasm/main.js:15]: ofs is the offset right and p is a pixel in the buffer? how about using the full names?

File: wasm/main.js [hunk 2]
```diff
@@ -7,21 +7,18 @@
...
+      pixel[ofs++] = (p >> 16) & 0xFF;
+      pixel[ofs++] = (p >> 8) & 0xFF;
+      pixel[ofs++] = p & 0xFF;
```
Comment (pfiaux) [inline wasm/main.js:18]: How about avoiding the magic numbers to make it more descriptive?
- 0xFF -> TWO_BYTE_MASK
- 16 -> SHIFT_TO_RED_BYTES
- 8 -> SHIFT_TO_GREEN_BYTES
- 255 -> FULL_OPACITY

File: wasm/main.js [hunk 3]
```diff
@@ -7,21 +7,18 @@
...
-    buffer.forEach((pixel) => {
-      if (pixel !== lastPixel) {
-        lastPixel = pixel;
-        ctx.fillStyle = '#' + pixel.toString(16).padStart(6, '0') + 'FF';
-      }
-      const x = offset % 320;
-      const y = parseInt(offset / 320, 10);
-      ctx.fillRect(x, y, 1, 1)
-      offset++;
+    const pixel = tempImage.data;
```
Comment (pfiaux) [inline wasm/main.js:13]: Should this be pixels or imagePixels?

File: wasm/main.js (hunk 2 above)
Comment (neophob) [reply wasm/main.js:18]: 0xFF is one byte ;) but good point!

**Pull request merged**